usaspending_api/agency/v2/views/tas_object_class_list.py

```diff
@@ -0,0 +1,151 @@
+from usaspending_api.references.models import ObjectClass
+from usaspending_api.references.models import RefProgramActivity
+
+
+class TASObjectClassList(PaginationMixin, AgencyBase):
+    """
+    Obtain the list of object classes for a specific agency's treasury
+    account (TAS).
+    """
+
+    endpoint_doc = "usaspending_api/api_contracts/contracts/v2/agency/treasury_account/tas/object_class.md"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.params_to_validate = ["fiscal_year", "filter"]
+
+    @property
+    def tas_rendering_label(self):
+        return self.kwargs["tas"]
+
+    @cache_response()
+    def get(self, request: Request, *args: Any, **kwargs: Any) -> Response:
+        self.sortable_columns = ["name", "obligated_amount", "gross_outlay_amount"]
+        self.default_sort_column = "obligated_amount"
+        order = self.pagination.sort_order == "desc"
+        results = list(self.get_object_class_list())
+        page_metadata = get_pagination_metadata(len(results), self.pagination.limit, self.pagination.page)
+        results = results[self.pagination.lower_limit : self.pagination.upper_limit]
+        pagination_limit_results = results[: self.pagination.limit]
+        pagination_limit_results = sorted(
+            pagination_limit_results, key=lambda x: x[self.pagination.sort_key], reverse=order
+        )
+        response_dict = {
+            "treasury_account_symbol": self.tas_rendering_label,
+            "fiscal_year": self.fiscal_year,
+            "page_metadata": page_metadata,
+            "results": pagination_limit_results,
+            "messages": self.standard_response_messages,
+        }
+        # There isn't a direct relationship between ObjectClass and RefProgramActivity
+        # That's why we opted to query for RefProgramActivity for each ObjectClass
+        for idx, object_class_row in enumerate(pagination_limit_results):
+            program_activity_results = self.get_program_activity_by_object_class_list(object_class_row["id"])
+            child_response_dict = self.format_program_activity_children_response(program_activity_results)
+            # "id" column should not be in the response
+            # Instead of adding another for loop to remove this column from the object class list results
+            # Just delete it during an already existing loop over the list
+            del response_dict["results"][idx]["id"]
+            child_response_dict = sorted(child_response_dict, key=lambda x: x[self.pagination.sort_key], reverse=order)
+            response_dict["results"][idx]["children"] = child_response_dict
+
+        return Response(response_dict)
+
+    def get_object_class_list(self) -> List[dict]:
+        submission_ids = get_latest_submission_ids_for_fiscal_year(self.fiscal_year)
+        filters = [
+            Q(financialaccountsbyprogramactivityobjectclass__submission_id__in=submission_ids),
```

Comment:
Not doubting that it works since you've done some testing. Just curious how the relationship between `ObjectClass` to `FinancialAccountsByProgramActivityObjectClass` is being resolved here. Same with `RefProgramActivity` later on.

Reply:
`FinancialAccountsByProgramActivityObjectClass` has a foreign key to `ObjectClass`. If I'm not mistaken Django will "automatically" join the needed relationships together. That's how object class is being pulled in here. There's also an established relationship between `RefProgramActivity` and `FinancialAccountsByProgramActivityObjectClass`

Reply:
Cool makes sense. I've seen us use `related_name` on the foreign key to create a back reference. Didn't realize Django created one by default even without the `related_name`.

Reply:
I guess it just uses the model name by default.